### PR TITLE
BigDecimal value fix in enums (#715)

### PR DESCRIPTION
**Pull request by Shohou**
a fix for issue #715. It works and test passes, but please check my small changes as I didn't work with Qute before and maybe there is a better solution

### CODE REVIEW

**Review by hbelmiro — APPROVED** [review]

**Comment by ricardozanini** [comment]
@fjtirado can you take a look since you introduced this fix?

**Comment by hbelmiro** [comment]
@fjtirado are you willing to review this PR? Otherwise I'll merge it.

File: client/deployment/src/main/resources/templates/libraries/microprofile/enumClass.qute [hunk 1]
```diff
@@ -7,7 +7,7 @@
         {#if e.withXml}
         {#for v in e.allowableValues.enumVars}@XmlEnumValue({#if v.isInteger || v.isDouble || v.isLong || v.isFloat}"{/if}{v.value}{#if v.isInteger || v.isDouble || v.isLong || v.isFloat}"{/if}) {v.name}({#if e.isEnum}{e.items.dataType}{#else}{e.dataType}{/if}.valueOf({v.value})){#if v_hasNext}, {#else}; {/if}{/for}
         {#else}
-        {#for v in e.allowableValues.enumVars}{v.name}({#if e.isContainer}{e.items.dataType}{#else}{e.dataType}{/if}.valueOf({v.value})){#if v_hasNext}, {#else};{/if}{/for}
+        {#for v in e.allowableValues.enumVars}{v.name}({#if eq e.dataType "BigDecimal"}{v.value}{#else}{#if e.isContainer}{e.items.dataType}{#else}{e.dataType}{/if}.valueOf({v.value}){/if}){#if v_hasNext}, {#else};{/if}{/for}
```
Comment (fjtirado) [inline client/deployment/src/main/resources/templates/libraries/microprofile/enumClass.qute:10]: Im wondering if we can somehow make this more "generic" and check if e.dataType is a number

Comment (fjtirado) [reply client/deployment/src/main/resources/templates/libraries/microprofile/enumClass.qute:10]: Maybe 
`  {#for v in e.allowableValues.enumVars}{v.name}({#if v.isNumeric}{v.value}{#else}{#if e.isContainer}{e.items.dataType}{#else}{e.dataType}{/if}.valueOf({v.value}){/if}){#if v_hasNext}, {#else};{/if}{/for}
`

**Comment by fjtirado** [comment]
@hbelmiro @Shohou I think `eq e.dataType "BigInteger"` should be replaced by `e.isNumeric`

**Comment by Shohou** [comment]
@fjtirado done

**Review by fjtirado — APPROVED** [review]

**Pull request merged**